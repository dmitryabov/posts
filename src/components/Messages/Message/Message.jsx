import React from 'react';
import style from './Message.module.css';


const Message = (props) => {
  
    return (
          <div className={style.container}>
            <p><span>{props.user.name}</span> {props.user.username}</p>
            <p>{props.message.title}</p>
            <p><span>{props.message.body}</span></p>
          </div>         
    )
}

export default Message;