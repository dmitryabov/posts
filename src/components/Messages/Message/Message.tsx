import React from 'react';
import style from './Message.module.css';
import {MessagesType} from '../../../redux/messages-reduser';


type PropsType = {
  message: MessagesType
  user: any
}

const Message: React.FC<PropsType> = (props) => {
    return (
          <div className={style.container}>
            <p><span>{props.user.name}</span> {props.user.username}</p>
            <p>{props.message.title}</p>
            <p><span>{props.message.body}</span></p>
          </div>         
    )
}

export default Message;