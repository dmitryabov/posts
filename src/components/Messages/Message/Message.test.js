import React from "react";
import renderer from "react-test-renderer";
import Message from "./Message";


const user =
{
  "id": 1,
  "name": "Leanne Graham",
  "username": "Bret",
  "email": "Sincere@april.biz",
  "address": {
    "street": "Kulas Light",
    "suite": "Apt. 556",
    "city": "Gwenborough",
    "zipcode": "92998-3874",
    "geo": {
      "lat": "-37.3159",
      "lng": "81.1496"
    }
  }
};

const message =
{
  "userId": 1,
  "id": 1,
  "title": "sunt aut facere repellat provident occaecati excepturi optio reprehenderit",
  "body": "quia et suscipit\nsuscipit recusandae consequuntur expedita et cum\nreprehenderit molestiae ut ut quas totam\nnostrum rerum est autem sunt rem eveniet architecto"
}


it(`Render Message`, () => {
  const tree = renderer
  .create(<Message
    message={message}
    user={user}
  />, {
    createNodeMock: ()=>{
      return {};
    }
  }).toJSON();

  expect(tree).toMatchSnapshot();
});
