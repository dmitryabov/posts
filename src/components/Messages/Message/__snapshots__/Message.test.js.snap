// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Message 1`] = `
<div
  className="container"
>
  <p>
    <span>
      Leanne Graham
    </span>
     
    Bret
  </p>
  <p>
    sunt aut facere repellat provident occaecati excepturi optio reprehenderit
  </p>
  <p>
    <span>
      quia et suscipit
suscipit recusandae consequuntur expedita et cum
reprehenderit molestiae ut ut quas totam
nostrum rerum est autem sunt rem eveniet architecto
    </span>
  </p>
</div>
`;

exports[`Render User 1`] = `
<div
  className="container"
>
  <p>
    <span>
      Leanne Graham
    </span>
     
    Bret
  </p>
  <p>
    sunt aut facere repellat provident occaecati excepturi optio reprehenderit
  </p>
  <p>
    <span>
      quia et suscipit
suscipit recusandae consequuntur expedita et cum
reprehenderit molestiae ut ut quas totam
nostrum rerum est autem sunt rem eveniet architecto
    </span>
  </p>
</div>
`;
